keystore_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestKeystore(t *testing.T) {
+	tfile, err := ioutil.TempFile("", "mendertest")
+	if err != nil {
+		t.Fatal("failed to create temporary file")
+	}
+	tname := tfile.Name()
+	tfile.Close()
+
+	k := NewKeystore()
+
+	err = k.Load("foo")
+	assert.Error(t, err)
+	assert.True(t, IsNoKeys(err))
+	assert.Nil(t, k.Private())
+
+	// try using temp file, this time we should get unmarhall/load
+	// error
+	err = k.Load(tname)
+	assert.Error(t, err)
+	assert.False(t, IsNoKeys(err))
+	assert.Nil(t, k.Private())
+
+	// not changing random source, so this is no expected to fail
+	assert.NoError(t, k.Generate())
+
+	assert.NotNil(t, k.Private())
+
+	// try to save in at location that triggers an error
+	assert.Error(t, k.Save("/foo"))
+
+	// try our temp location now
+	assert.NoError(t, k.Save(tname))
+
+	// we should be able to load a saved key
+	assert.NoError(t, k.Load(tname))
+
+	os.Remove(tname)
```

Comment:
Shouldn't we have `defer os.Remove(tfile.Name()) just after checking that file was created?